ibis/expr/operations/udf.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import functools
+import inspect
+import itertools
+import typing
+from typing import TYPE_CHECKING, Any, Callable, TypeVar
+
+from public import public
+
+import ibis.common.exceptions as exc
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.rules as rlz
+from ibis import util
+from ibis.common.annotations import attribute
+from ibis.common.collections import FrozenDict
+
+if TYPE_CHECKING:
+    import ibis.expr.types as ir
+
+
+@public
+class ScalarUDF(ops.Value):
+    @attribute.default
+    def output_shape(self):
```

Comment:
Is it equivalent with `rlz.shape_like("args")`?

Reply:
I think so!